admin/tools/docker/replication/deployment/ncsa/env.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# This file is part of lsst/qserv.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# This script is supposed to be sourced from a client script in order
+# to set up proper values of the corresponding parameters
+
+basedir=$(dirname $0)
+if [ -z "$basedir" ] || [ "$0" = "bash" ]; then
+    (>&2 echo "error: variable 'basedir' is not defined")
+    exit 1 
```

Comment:
This exist is done on purpose because env.sh is sourced from the installation management scripts status/start/stop/run scripts which are run from a new shell. So, terminating that shell is the right thing to do here.

Reply:
I think this is bad design, it means that both this script and its caller make a lot of assumptions about each other behavior. Caller should do proper error handling, and this script needs to be a bit more friendly so it can be called not only from one particular place.

Reply:
Still if you insist on this behavior put a big banner in large red letters saying that this script will terminate a script which sources it. 😈

Reply:
The truth is that these scripts (as I've already indicated tis earlier) is just  temporary solution before everything will get wrapped into the k8. There are just 5 simple (trivial) clients of this particular (environment setup) script. And I see no compelling reason to over-design tis implementation juts in case or in an anticipation of some heavy development utilizing this script. Hence I'm confident that terminating a caller is still the right ting to do.